multimodal/src/autogluon/multimodal/data/dataset_mmlab/multi_image_mix_dataset.py

```diff
@@ -17,834 +17,823 @@
 
 try:
     import mmcv
-    from mmcv.utils import Config as MMCVConfig
-except:
-    MMCVConfig = None
-
-try:
     from mmcv.transforms import BaseTransform
     from mmcv.transforms.utils import cache_randomness
-except:
-    pass
-
-try:
+    from mmdet.structures.bbox import autocast_box_type
+    from mmdet.utils import log_img_scale
+    from mmengine.config import Config as MMConfig
     from mmengine.dataset import BaseDataset
-except:
-    pass
 
-try:
-    from mmdet.structures.bbox import autocast_box_type
-    from mmdet.utils import log_img_scale  # inline import to avoid mmdet uninstall error for other tasks
-except:
-    pass
-
-
-class MultiImageMixDataset(torch.utils.data.Dataset):
-    """
-    A Pytorch DataSet class to process a multimodal pd.DataFrame. It first uses a preprocessor to
-    produce model-agnostic features. Then, each processor prepares customized data for one modality
-    per model. For code simplicity, here we treat ground-truth label as one modality. This class is
-    independent of specific data modalities and models.
-    """
-
-    def __init__(
-        self,
-        data: pd.DataFrame,
-        preprocessor: List[MultiModalFeaturePreprocessor],
-        processors: List[dict],
-        model_config: MMCVConfig,
-        id_mappings: Optional[Union[Dict[str, Dict], Dict[str, pd.Series]]] = None,
-        is_training: bool = False,
-    ):
+    class MultiImageMixDataset(torch.utils.data.Dataset):
```

Comment:
Instead of putting the entire code of the file in a try/except block, create a new file `_multi_image_mix_dataset.py` with the original code and no try/except. Then in this file, simply import the classes from that file in a try/except.

```python3
try:
    from ._multi_image_mix_dataset import MultiImageMixDataset
except ImportError as e:
    MultiImageMixDataset = None
```

This will be much cleaner.

Reply:
Oh sure, or I can also add the try block in __init__?

Reply:
If you want to add to init, you can do that, but the file with the code source should not have the try/except block for the class definitions

Reply:
Addressed in #3277